src/System.Private.Xml.Linq/src/System/Xml/Linq/XHashtable.cs

```diff
@@ -397,22 +398,9 @@ private bool FindEntry(int hashCode, string key, int index, int count, ref int e
             /// </summary>
             private static int ComputeHashCode(string key, int index, int count)
             {
-                int hashCode = StartingHash;
-                int end = index + count;
                 Debug.Assert(key != null, "key should have been checked previously for null");
-
-                // Hash the key
-                for (int i = index; i < end; i++)
-                    unchecked
-                    {
-                        hashCode += (hashCode << 7) ^ key[i];
-                    }
-
-                // Mix up hash code a bit more and clear the sign bit.  This code was taken from NameTable.cs in System.Xml.
-                hashCode -= hashCode >> 17;
-                hashCode -= hashCode >> 11;
-                hashCode -= hashCode >> 5;
-                return hashCode & 0x7FFFFFFF;
+                ReadOnlySpan<byte> bytes = MemoryMarshal.AsBytes(key.AsSpan(index, index + count));
```

Comment:
https://source.dot.net/#System.Private.CoreLib/shared/System/MemoryExtensions.Fast.cs,e3849b379849b4a4

The arguments to AsSpan are `start` and `length`, which should just be `index` and `count`... so adding index to count (to produce `end`) seems wrong.

Reply:
uh, I thought it is end index